internal/fsapi/file.go

```diff
@@ -371,12 +366,155 @@ type File interface {
 
 	// Close closes the underlying file.
 	//
+	// A zero syscall.Errno is returned if unimplemented or success.
+	//
+	// # Notes
+	//
+	//   - This is like syscall.Close and `close` in POSIX. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/close.html
+	Close() syscall.Errno
+}
+
+// Readdir is the result of File.Readdir and it is a cursor over a directory.
+//
+// # Errors
+//
+// All methods that can return an error return a syscall.Errno, which is zero
+// on success.
+//
+// # Notes
+//
+//   - Implementations are allowed to keep a buffer or a sliding window of values
+//     over a directory. Because reading in batches requires to hold a reference
+//     to an open directory, callers should pay extra care to make sure
+//     that such references are released, by invoking Close.
+//   - This is especially important on the Windows platform, where keeping
+//     a file handle open may prevent concurrent processes from accessing
+//     (e.g. deleting) the underlying directory. This may conflict with cleanup
+//     procedures (such as `testing/TB.Cleanup()`).
+type Readdir interface {
+	// Skip is equivalent to calling Next n times.
+	Skip(n uint64)
```

Comment:
locally I only see this used in tests. Can we delete anything only used in tests for something similar inlined?

(ack this is not something you added, rather moved)